### ISSUE TITLE
Incorporate CloudFront and DNS into the CloudFormation Stack

### ISSUE THREAD

**Issue by freeformflow** [pandastrike/panda-sky#19]
Currently the configuration for the API's custom URL, a combination of a CloudFront distribution and DNS records, is set independently of the CloudFormation stack.  That makes its behavior more annoying to deal with and less predicable when it comes to tear down.

Refactor the functions that assemble the CloudFront and DNS config and dump it all into the deployment's stack.

**Comment by freeformflow** [comment]
@dyoder, I just remembered why I didn't do it this way to begin with when I did things better with Panda Sky.  CloudFront distros setup by CloudFormation **DO NOT** support ACM certs natively (ugh).  So to get around it, I'd have to create a custom CloudFormation Resource that would then ping a Lambda that would use the AWS SDK to fetch the cert's ARN.

So, not only would I *still* be using the SDK, but it's waaaay more ugly, haha.  So...  I guess this will need to be put on hold.  I still need to debug #20, which will take some time.   That being said, AWS could release the fix for this tomorrow, haha.
